### PR TITLE
Feature thirdparty automatization

### PR DESCRIPTION
**Pull request by Quentin01**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -137,7 +138,21 @@ matrix:
 
 # scripts that run after cloning repository
 install:
-  - git submodule update --init --recursive
+  - ps: |
+      if((Test-Path thirdparty) -eq 0)
+      {
+        cd ..
```
Comment (saveman71) [inline appveyor.yml:144]: Maybe instead
```
git clone https://github.com/Lugdunum3D/Lugdunum-ThirdParty.git ../Lugdunum-ThirdParty
pip install -r ../Lugdunum-ThirdParty/requirements.txt
python ../Lugdunum-ThirdParty/build.py -vvv --path thirdparty thirdparty.yml
```

I'm not sure if the pip line works, but I don't see why not

File: cmake/modules/FindShaderc.cmake [hunk 2]
```diff
@@ -16,14 +16,14 @@ find_path(SHADERC_INCLUDE_DIR
 find_library(SHADERC_LIBRARY_DEBUG
     NAMES shaderc_combined-d
     PATHS $ENV{SHADERC_ROOT} ${SHADERC_ROOT}
-    PATH_SUFFIXES lib64 lib/${ANDROID_ABI}
+    PATH_SUFFIXES lib lib64 lib/${ANDROID_ABI}
```
Comment (saveman71) [inline cmake/modules/FindShaderc.cmake:19]: Thank

you

:+1:

File: include/lug/Graphics/Vulkan/Render/Pipeline.hpp [hunk 3]
```diff
@@ -121,7 +121,7 @@ class LUG_GRAPHICS_API Pipeline : public Resource {
     };
 
 public:
-    class ShaderBuilder {
+    class LUG_GRAPHICS_API ShaderBuilder {
```
Comment (saveman71) [inline include/lug/Graphics/Vulkan/Render/Pipeline.hpp:124]: Should this be here?

File: src/lug/Graphics/Vulkan/Render/Pipeline.cpp [hunk 4]
```diff
@@ -109,7 +109,7 @@ bool Pipeline::init() {
 
             std::vector<VkDescriptorSetLayoutBinding> bindings = {cameraBinding};
 
-            descriptorSetLayouts.push_back(API::DescriptorSetLayout::create(&_renderer.getDevice(), bindings.data(), bindings.size()));
```
Comment (saveman71) [inline src/lug/Graphics/Vulkan/Render/Pipeline.cpp:112]: Same here, should those modifications be in this PR at all?

File: include/lug/Graphics/Vulkan/Render/Pipeline.hpp (hunk 3 above)
Comment (Quentin01) [reply include/lug/Graphics/Vulkan/Render/Pipeline.hpp:124]: Without it, the build on Appveyor fails

File: src/lug/Graphics/Vulkan/Render/Pipeline.cpp (hunk 4 above)
Comment (Quentin01) [reply src/lug/Graphics/Vulkan/Render/Pipeline.cpp:112]: Same, without it, the build on Appveyor fails

**Review by saveman71 — APPROVED** [review]

**Pull request merged**